src/systems/binds.cpp

```diff
@@ -0,0 +1,191 @@
+    s_pressed[1] =
+        pad::button_pressed(mkb::PAD_BUTTON_B, true) || pad::button_down(mkb::PAD_BUTTON_B, true);
+    s_pressed[2] =
+        pad::button_pressed(mkb::PAD_BUTTON_X, true) || pad::button_down(mkb::PAD_BUTTON_X, true);
+    s_pressed[3] =
+        pad::button_pressed(mkb::PAD_BUTTON_Y, true) || pad::button_down(mkb::PAD_BUTTON_Y, true);
+    s_pressed[4] =
+        pad::button_pressed(mkb::PAD_TRIGGER_L, true) || pad::button_down(mkb::PAD_TRIGGER_L, true);
+    s_pressed[5] =
+        pad::button_pressed(mkb::PAD_TRIGGER_R, true) || pad::button_down(mkb::PAD_TRIGGER_R, true);
+    s_pressed[6] =
+        pad::button_pressed(mkb::PAD_TRIGGER_Z, true) || pad::button_down(mkb::PAD_TRIGGER_Z, true);
+    s_pressed[7] =
+        pad::button_pressed(mkb::PAD_BUTTON_UP, true) || pad::button_down(mkb::PAD_BUTTON_UP, true);
+    s_pressed[8] = pad::button_pressed(mkb::PAD_BUTTON_DOWN, true) ||
+                   pad::button_down(mkb::PAD_BUTTON_DOWN, true);
+    s_pressed[9] = pad::button_pressed(mkb::PAD_BUTTON_LEFT, true) ||
+                   pad::button_down(mkb::PAD_BUTTON_LEFT, true);
+    s_pressed[10] = pad::button_pressed(mkb::PAD_BUTTON_RIGHT, true) ||
+                    pad::button_down(mkb::PAD_BUTTON_RIGHT, true);
+    s_pressed[11] = pad::button_pressed(mkb::PAD_BUTTON_START, true) ||
+                    pad::button_down(mkb::PAD_BUTTON_START, true);
+}
+
+static bool is_num_pressed(int num) {
+    if (num >= 0 && num < 12) {
+        return s_pressed[num];
+    }
+    return false;
+}
+
+static mkb::PadDigitalInput num_to_pad(int num) {
+    switch (num) {
+        case 0: {
+            return mkb::PAD_BUTTON_A;
+        }
+        case 1: {
+            return mkb::PAD_BUTTON_B;
+        }
+        case 2: {
+            return mkb::PAD_BUTTON_X;
+        }
+        case 3: {
+            return mkb::PAD_BUTTON_Y;
+        }
+        case 4: {
+            return mkb::PAD_TRIGGER_L;
+        }
+        case 5: {
+            return mkb::PAD_TRIGGER_R;
+        }
+        case 6: {
+            return mkb::PAD_TRIGGER_Z;
+        }
+        case 7: {
+            return mkb::PAD_BUTTON_UP;
+        }
+        case 8: {
+            return mkb::PAD_BUTTON_DOWN;
+        }
+        case 9: {
+            return mkb::PAD_BUTTON_LEFT;
+        }
+        case 10: {
+            return mkb::PAD_BUTTON_RIGHT;
+        }
+        case 11: {
+            return mkb::PAD_BUTTON_START;
+        }
+        default: {
+            return 0;
+        }
+    }
+}
+
+static void encode_bind() {
+    u8 encoding = 0;
+    encoding += s_prev_pressed[0];
+
+    if (s_prev_pressed[1] != 255) {
+        encoding += s_prev_pressed[1] * 12;
+    } else {
+        encoding += s_prev_pressed[0] * 12;
+    }
+
+    s_encoding = encoding;
+}
+
+void init() {}
+
+void tick() {
+    get_button_values();
+
+    u8 pressed_count = 0;
+    s_current_pressed[0] = 255;
+    s_current_pressed[1] = 255;
```

Comment:
This can be a local variable

Reply:
I meant `s_current_pressed` can be stack allocated in the function instead of being a static variable, but the INVALID constant is welcome too